ethicml/utility/lookup.py

```diff
@@ -0,0 +1,25 @@
+"""Lookup tables / Swutch statements for project level objects."""
+
+from typing import Dict
+
+from ethicml.data import Dataset, adult, compas, credit, german, nonbinary_toy, sqf, toy
+
+__all__ = ["get_dataset_obj_by_name"]
+
+
+def get_dataset_obj_by_name(name: str) -> Dataset:
+    """Given a dataset name, get the corresponding dataset object."""
+    lookup: Dict[str, Dataset] = {
```

Comment:
```suggestion
    lookup: Dict[str, Callable[[], Dataset] = {
```

We could also make it so that we just look up the function and only then call it to get the dataset object.

Reply:
I'm not sure as this isn't something that's used all that often. In fact, given the recent changes to Dataset, I'm not sure if this is still needed, but it probably falls out of the scope of this PR to check that

Reply:
yeah it was probably a micro optimization